chainer/functions/math/trigonometric.py

```diff
@@ -67,3 +67,36 @@ def backward_gpu(self, x, gy):
 def cos(x):
     """Elementwise cos function."""
     return Cos()(x)
+
+
+_preamble = '''
+template <typename T> __device__ T sqr(T x) { return x * x; }
+'''
+
+
+class Tan(function.Function):
+
+    @property
```

Comment:
Yes, and it is supposed to be one of reasonable options to prevent users from assigning new labels. Of course, leaving it allowed for users would be another option. This time I followed other functions.